core/src/test/scala/io/finch/FinchSpec.scala

```diff
@@ -304,12 +306,12 @@ trait FinchSpec extends AnyFlatSpec with Matchers with Checkers with AllInstance
     *
     * We attempt to verify that two endpoints are the same by applying them to a fixed number of randomly generated inputs.
     */
-  implicit def eqEndpoint[F[_]: Effect, A: Eq]: Eq[Endpoint[F, A]] = new Eq[Endpoint[F, A]] {
+  implicit def eqEndpoint[F[_]: Sync, A: Eq](implicit dispatcher: Dispatcher[F]): Eq[Endpoint[F, A]] = new Eq[Endpoint[F, A]] {
```

Comment:
Oh this eq methods need more attention. Maybe we could move these under missingInstances that's instantiated around the existing dispatcher.

Reply:
Existing dispatcher is already set on IO (and eq is used based on higher kind). I'll let it that way until next PR 🚀